Release 11-12-2024

Junjiequan opened the pull request:
<!--
Follow semantic-release guidelines for the PR title, which is used in the changelog.

Title should follow the format `<type>(<scope>): <subject>`, where
- Type is one of: build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test|BREAKING CHANGE
- Scope (optional) describes the place of the change (eg a particular milestone) and is usually omitted
- subject should be a non-capitalized one-line description in present imperative tense and not ending with a period

See https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines for more details.
-->

## Description
<!-- Short description of the pull request -->

## Motivation
<!-- Background on use case, changes needed -->

## Fixes
<!-- Please provide a list of the issues fixed by this PR -->

* Bug fixed (#X)

## Changes:
<!-- Please provide a list of the changes implemented by this PR -->

* changes made

## Tests included

- [ ] Included for each change/fix?
- [ ] Passing? <!-- Merge will not be approved unless tests pass -->

## Documentation
- [ ] swagger documentation updated (required for API changes)
- [ ] official documentation updated

### official documentation info
<!-- If you have updated the official documentation, please provide PR # and URL of the updated pages -->

HayenNico committed:
implement configuration options for custom dataset types

HayenNico committed:
update dataset schema and module to allow custom type names

HayenNico committed:
update dataset create endpoint to accept custom typed datasets

HayenNico committed:
fix: dataset create endpoint details

HayenNico committed:
add: runNumber property in update-dataset DTO

HayenNico committed:
update datasets IsValid endpoint for custom dataset types

HayenNico committed:
allow custom dataset types in patch endpoint

HayenNico committed:
allow custom dataset types in put/update endpoint

HayenNico committed:
appease eslint

HayenNico committed:
more eslint fixes

HayenNico committed:
even more eslint fixes

HayenNico committed:
adjust test workflow for new config options

HayenNico committed:
add test data, remove dataset type check in module

HayenNico committed:
add new api tests for custom typed datasets

HayenNico committed:
revert change in package.json

HayenNico committed:
refactor switch statements in separate method

HayenNico committed:
remove unused imports in dataset module

HayenNico committed:
eslint fix

HayenNico committed:
revert refactor of switch statements, fix wrong type selection for put endpoint

HayenNico committed:
update .env.example to include new field

martin-trajanovski committed:
Merge branch 'master' into feature-custom-dataset-types

martin-trajanovski committed:
feat: migrate dataset principalInvestigator to principalInvestigators

nitrosx committed:
Merge branch 'master' into migrate-principal-investigator

nitrosx committed:
Merge branch 'master' into feature-custom-dataset-types

HayenNico committed:
fix: lint

HayenNico committed:
add back isValid functionality for custom types

HayenNico committed:
Merge pull request #1532 from SciCatProject/feature-custom-dataset-types

feat: allow custom dataset types defined in configuration

Junjiequan committed:
chore: revert packge-lock changes and update minor versions of the packages

dependabot[bot] committed:
build(deps): bump openid-client from 5.7.0 to 5.7.1

Bumps [openid-client](https://github.com/panva/openid-client) from 5.7.0 to 5.7.1.
- [Release notes](https://github.com/panva/openid-client/releases)
- [Changelog](https://github.com/panva/openid-client/blob/v5.7.1/CHANGELOG.md)
- [Commits](https://github.com/panva/openid-client/compare/v5.7.0...v5.7.1)

---
updated-dependencies:
- dependency-name: openid-client
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1551 from SciCatProject/dependabot/npm_and_yarn/openid-client-5.7.1

build(deps): bump openid-client from 5.7.0 to 5.7.1

dependabot[bot] committed:
build(deps-dev): bump @types/express-session from 1.18.0 to 1.18.1

Bumps [@types/express-session](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express-session) from 1.18.0 to 1.18.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/express-session)

---
updated-dependencies:
- dependency-name: "@types/express-session"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1547 from SciCatProject/dependabot/npm_and_yarn/types/express-session-1.18.1

build(deps-dev): bump @types/express-session from 1.18.0 to 1.18.1

dependabot[bot] committed:
build(deps-dev): bump @types/node from 22.9.0 to 22.10.1

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 22.9.0 to 22.10.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1546 from SciCatProject/dependabot/npm_and_yarn/types/node-22.10.1

build(deps-dev): bump @types/node from 22.9.0 to 22.10.1

dependabot[bot] committed:
build(deps-dev): bump @faker-js/faker from 9.2.0 to 9.3.0

Bumps [@faker-js/faker](https://github.com/faker-js/faker) from 9.2.0 to 9.3.0.
- [Release notes](https://github.com/faker-js/faker/releases)
- [Changelog](https://github.com/faker-js/faker/blob/next/CHANGELOG.md)
- [Commits](https://github.com/faker-js/faker/compare/v9.2.0...v9.3.0)

---
updated-dependencies:
- dependency-name: "@faker-js/faker"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1550 from SciCatProject/dependabot/npm_and_yarn/faker-js/faker-9.3.0

build(deps-dev): bump @faker-js/faker from 9.2.0 to 9.3.0

dependabot[bot] committed:
build(deps): bump @nestjs/axios from 3.1.2 to 3.1.3

Bumps [@nestjs/axios](https://github.com/nestjs/axios) from 3.1.2 to 3.1.3.
- [Release notes](https://github.com/nestjs/axios/releases)
- [Changelog](https://github.com/nestjs/axios/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/axios/compare/3.1.2...3.1.3)

---
updated-dependencies:
- dependency-name: "@nestjs/axios"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1549 from SciCatProject/dependabot/npm_and_yarn/nestjs/axios-3.1.3

build(deps): bump @nestjs/axios from 3.1.2 to 3.1.3

dependabot[bot] committed:
build(deps): bump mongoose from 8.8.3 to 8.8.4

Bumps [mongoose](https://github.com/Automattic/mongoose) from 8.8.3 to 8.8.4.
- [Release notes](https://github.com/Automattic/mongoose/releases)
- [Changelog](https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Automattic/mongoose/compare/8.8.3...8.8.4)

---
updated-dependencies:
- dependency-name: mongoose
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1552 from SciCatProject/dependabot/npm_and_yarn/mongoose-8.8.4

build(deps): bump mongoose from 8.8.3 to 8.8.4

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 8.16.0 to 8.17.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.17.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1545 from SciCatProject/dependabot/npm_and_yarn/typescript-eslint/eslint-plugin-8.17.0

build(deps-dev): bump @typescript-eslint/eslint-plugin from 8.16.0 to 8.17.0

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 8.16.0 to 8.17.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.16.0 to 8.17.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.17.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1548 from SciCatProject/dependabot/npm_and_yarn/typescript-eslint/parser-8.17.0

build(deps-dev): bump @typescript-eslint/parser from 8.16.0 to 8.17.0

dependabot[bot] committed:
build(deps-dev): bump @nestjs/testing from 10.4.1 to 10.4.15

Bumps [@nestjs/testing](https://github.com/nestjs/nest/tree/HEAD/packages/testing) from 10.4.1 to 10.4.15.
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/commits/v10.4.15/packages/testing)

---
updated-dependencies:
- dependency-name: "@nestjs/testing"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1553 from SciCatProject/dependabot/npm_and_yarn/nestjs/testing-10.4.15

build(deps-dev): bump @nestjs/testing from 10.4.1 to 10.4.15

dependabot[bot] committed:
build(deps-dev): bump @nestjs/cli from 10.4.7 to 10.4.9

Bumps [@nestjs/cli](https://github.com/nestjs/nest-cli) from 10.4.7 to 10.4.9.
- [Release notes](https://github.com/nestjs/nest-cli/releases)
- [Changelog](https://github.com/nestjs/nest-cli/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/nest-cli/compare/10.4.7...10.4.9)

---
updated-dependencies:
- dependency-name: "@nestjs/cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1554 from SciCatProject/dependabot/npm_and_yarn/nestjs/cli-10.4.9

build(deps-dev): bump @nestjs/cli from 10.4.7 to 10.4.9

Junjiequan committed:
fix: update schema validation from oneOf to anyOf for dataset creation and updates (#1544)

Co-authored-by: Martin <martin.trajanovski@gmail.com>

dependabot[bot] committed:
build(deps): bump path-to-regexp, @nestjs/platform-express and express

Bumps [path-to-regexp](https://github.com/pillarjs/path-to-regexp) to 3.3.0 and updates ancestor dependencies [path-to-regexp](https://github.com/pillarjs/path-to-regexp), [@nestjs/platform-express](https://github.com/nestjs/nest/tree/HEAD/packages/platform-express) and [express](https://github.com/expressjs/express). These dependencies need to be updated together.


Updates `path-to-regexp` from 0.1.10 to 3.3.0
- [Release notes](https://github.com/pillarjs/path-to-regexp/releases)
- [Changelog](https://github.com/pillarjs/path-to-regexp/blob/master/History.md)
- [Commits](https://github.com/pillarjs/path-to-regexp/compare/v0.1.10...v3.3.0)

Updates `@nestjs/platform-express` from 10.4.7 to 10.4.15
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/commits/v10.4.15/packages/platform-express)

Updates `express` from 4.21.1 to 4.21.2
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/4.21.2/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.21.1...4.21.2)

---
updated-dependencies:
- dependency-name: path-to-regexp
  dependency-type: indirect
- dependency-name: "@nestjs/platform-express"
  dependency-type: direct:production
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1555 from SciCatProject/dependabot/npm_and_yarn/multi-5c14921503

build(deps): bump path-to-regexp, @nestjs/platform-express and express

martin-trajanovski committed:
resolve merge conflicts and update description

martin-trajanovski committed:
try to fix some validation and failing tests

martin-trajanovski committed:
Merge pull request #1536 from SciCatProject/migrate-principal-investigator

feat: migrate dataset principalInvestigator to principalInvestigators

Junjiequan committed:
feat: add support for typescript-fetch SDK and update typescript-angular configuration (#1556)

* feat: add support for typescript-fetch SDK and update typescript-angular configuration

* correct the sdk name

* handle extra spaces

sbliven commented:
Can you link the changelog and documentation for this release? Since it includes datasets schema changes I think we should summarize the changes.

dependabot[bot] committed:
build(deps-dev): bump @types/nodemailer from 6.4.15 to 6.4.17

Bumps [@types/nodemailer](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/nodemailer) from 6.4.15 to 6.4.17.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/nodemailer)

---
updated-dependencies:
- dependency-name: "@types/nodemailer"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1565 from SciCatProject/dependabot/npm_and_yarn/types/nodemailer-6.4.17

build(deps-dev): bump @types/nodemailer from 6.4.15 to 6.4.17

dependabot[bot] committed:
build(deps-dev): bump concurrently from 9.0.0 to 9.1.0

Bumps [concurrently](https://github.com/open-cli-tools/concurrently) from 9.0.0 to 9.1.0.
- [Release notes](https://github.com/open-cli-tools/concurrently/releases)
- [Commits](https://github.com/open-cli-tools/concurrently/compare/v9.0.0...v9.1.0)

---
updated-dependencies:
- dependency-name: concurrently
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1567 from SciCatProject/dependabot/npm_and_yarn/concurrently-9.1.0

build(deps-dev): bump concurrently from 9.0.0 to 9.1.0

dependabot[bot] committed:
build(deps): bump mathjs from 14.0.0 to 14.0.1

Bumps [mathjs](https://github.com/josdejong/mathjs) from 14.0.0 to 14.0.1.
- [Changelog](https://github.com/josdejong/mathjs/blob/develop/HISTORY.md)
- [Commits](https://github.com/josdejong/mathjs/compare/v14.0.0...v14.0.1)

---
updated-dependencies:
- dependency-name: mathjs
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1573 from SciCatProject/dependabot/npm_and_yarn/mathjs-14.0.1

build(deps): bump mathjs from 14.0.0 to 14.0.1

dependabot[bot] committed:
build(deps-dev): bump @stylistic/eslint-plugin from 2.11.0 to 2.12.1

Bumps [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic/tree/HEAD/packages/eslint-plugin) from 2.11.0 to 2.12.1.
- [Release notes](https://github.com/eslint-stylistic/eslint-stylistic/releases)
- [Changelog](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint-stylistic/eslint-stylistic/commits/v2.12.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@stylistic/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1568 from SciCatProject/dependabot/npm_and_yarn/stylistic/eslint-plugin-2.12.1

build(deps-dev): bump @stylistic/eslint-plugin from 2.11.0 to 2.12.1

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 8.17.0 to 8.18.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.17.0 to 8.18.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.18.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1569 from SciCatProject/dependabot/npm_and_yarn/typescript-eslint/parser-8.18.0

build(deps-dev): bump @typescript-eslint/parser from 8.17.0 to 8.18.0

dependabot[bot] committed:
build(deps-dev): bump eslint from 9.15.0 to 9.17.0

Bumps [eslint](https://github.com/eslint/eslint) from 9.15.0 to 9.17.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.15.0...v9.17.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1566 from SciCatProject/dependabot/npm_and_yarn/eslint-9.17.0

build(deps-dev): bump eslint from 9.15.0 to 9.17.0

dependabot[bot] committed:
build(deps): bump @nestjs/elasticsearch from 10.0.1 to 10.0.2

Bumps @nestjs/elasticsearch from 10.0.1 to 10.0.2.

---
updated-dependencies:
- dependency-name: "@nestjs/elasticsearch"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1570 from SciCatProject/dependabot/npm_and_yarn/nestjs/elasticsearch-10.0.2

build(deps): bump @nestjs/elasticsearch from 10.0.1 to 10.0.2

Junjiequan commented:
> Can you link the changelog and documentation for this release? Since it includes datasets schema changes I think we should summarize the changes.

I will postpone this PR untill the project leader is back. I also feel like we (ESS) should give a convincing documentation/reason about the schema changes as it might be breaking changes for some users.

dependabot[bot] committed:
build(deps-dev): bump @types/node-fetch from 2.6.11 to 2.6.12

Bumps [@types/node-fetch](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node-fetch) from 2.6.11 to 2.6.12.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node-fetch)

---
updated-dependencies:
- dependency-name: "@types/node-fetch"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1583 from SciCatProject/dependabot/npm_and_yarn/types/node-fetch-2.6.12

build(deps-dev): bump @types/node-fetch from 2.6.11 to 2.6.12

dependabot[bot] committed:
build(deps-dev): bump @types/node from 22.10.1 to 22.10.2

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 22.10.1 to 22.10.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1571 from SciCatProject/dependabot/npm_and_yarn/types/node-22.10.2

build(deps-dev): bump @types/node from 22.10.1 to 22.10.2

dependabot[bot] committed:
build(deps): bump @nestjs/core from 10.4.6 to 10.4.15

Bumps [@nestjs/core](https://github.com/nestjs/nest/tree/HEAD/packages/core) from 10.4.6 to 10.4.15.
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/commits/v10.4.15/packages/core)

---
updated-dependencies:
- dependency-name: "@nestjs/core"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1585 from SciCatProject/dependabot/npm_and_yarn/nestjs/core-10.4.15

build(deps): bump @nestjs/core from 10.4.6 to 10.4.15

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 8.17.0 to 8.18.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.18.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1582 from SciCatProject/dependabot/npm_and_yarn/typescript-eslint/eslint-plugin-8.18.1

build(deps-dev): bump @typescript-eslint/eslint-plugin from 8.17.0 to 8.18.1

dependabot[bot] committed:
build(deps): bump @elastic/elasticsearch from 8.15.0 to 8.17.0

Bumps [@elastic/elasticsearch](https://github.com/elastic/elasticsearch-js) from 8.15.0 to 8.17.0.
- [Release notes](https://github.com/elastic/elasticsearch-js/releases)
- [Changelog](https://github.com/elastic/elasticsearch-js/blob/main/docs/changelog.asciidoc)
- [Commits](https://github.com/elastic/elasticsearch-js/compare/v8.15.0...v8.17.0)

---
updated-dependencies:
- dependency-name: "@elastic/elasticsearch"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1586 from SciCatProject/dependabot/npm_and_yarn/elastic/elasticsearch-8.17.0

build(deps): bump @elastic/elasticsearch from 8.15.0 to 8.17.0

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 8.18.0 to 8.18.1

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.18.0 to 8.18.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.18.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1584 from SciCatProject/dependabot/npm_and_yarn/typescript-eslint/parser-8.18.1

build(deps-dev): bump @typescript-eslint/parser from 8.18.0 to 8.18.1

dependabot[bot] committed:
build(deps): bump dotenv from 16.4.5 to 16.4.7

Bumps [dotenv](https://github.com/motdotla/dotenv) from 16.4.5 to 16.4.7.
- [Changelog](https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md)
- [Commits](https://github.com/motdotla/dotenv/compare/v16.4.5...v16.4.7)

---
updated-dependencies:
- dependency-name: dotenv
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1587 from SciCatProject/dependabot/npm_and_yarn/dotenv-16.4.7

build(deps): bump dotenv from 16.4.5 to 16.4.7

dependabot[bot] committed:
build(deps-dev): bump globals from 15.12.0 to 15.14.0

Bumps [globals](https://github.com/sindresorhus/globals) from 15.12.0 to 15.14.0.
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v15.12.0...v15.14.0)

---
updated-dependencies:
- dependency-name: globals
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1588 from SciCatProject/dependabot/npm_and_yarn/globals-15.14.0

build(deps-dev): bump globals from 15.12.0 to 15.14.0

dependabot[bot] committed:
build(deps-dev): bump prettier from 3.3.3 to 3.4.2

Bumps [prettier](https://github.com/prettier/prettier) from 3.3.3 to 3.4.2.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/3.3.3...3.4.2)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1589 from SciCatProject/dependabot/npm_and_yarn/prettier-3.4.2

build(deps-dev): bump prettier from 3.3.3 to 3.4.2

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 8.18.1 to 8.18.2

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 8.18.1 to 8.18.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.18.2/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1591 from SciCatProject/dependabot/npm_and_yarn/typescript-eslint/parser-8.18.2

build(deps-dev): bump @typescript-eslint/parser from 8.18.1 to 8.18.2

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 8.18.1 to 8.18.2.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.18.2/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1596 from SciCatProject/dependabot/npm_and_yarn/typescript-eslint/eslint-plugin-8.18.2

build(deps-dev): bump @typescript-eslint/eslint-plugin from 8.18.1 to 8.18.2

dependabot[bot] committed:
build(deps-dev): bump @nestjs/schematics from 10.1.4 to 10.2.3

Bumps [@nestjs/schematics](https://github.com/nestjs/schematics) from 10.1.4 to 10.2.3.
- [Release notes](https://github.com/nestjs/schematics/releases)
- [Changelog](https://github.com/nestjs/schematics/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/schematics/compare/10.1.4...10.2.3)

---
updated-dependencies:
- dependency-name: "@nestjs/schematics"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1592 from SciCatProject/dependabot/npm_and_yarn/nestjs/schematics-10.2.3

build(deps-dev): bump @nestjs/schematics from 10.1.4 to 10.2.3

dependabot[bot] committed:
build(deps-dev): bump @types/lodash from 4.17.12 to 4.17.13

Bumps [@types/lodash](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/lodash) from 4.17.12 to 4.17.13.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/lodash)

---
updated-dependencies:
- dependency-name: "@types/lodash"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1594 from SciCatProject/dependabot/npm_and_yarn/types/lodash-4.17.13

build(deps-dev): bump @types/lodash from 4.17.12 to 4.17.13

dependabot[bot] committed:
build(deps-dev): bump concurrently from 9.1.0 to 9.1.1

Bumps [concurrently](https://github.com/open-cli-tools/concurrently) from 9.1.0 to 9.1.1.
- [Release notes](https://github.com/open-cli-tools/concurrently/releases)
- [Commits](https://github.com/open-cli-tools/concurrently/compare/v9.1.0...v9.1.1)

---
updated-dependencies:
- dependency-name: concurrently
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1593 from SciCatProject/dependabot/npm_and_yarn/concurrently-9.1.1

build(deps-dev): bump concurrently from 9.1.0 to 9.1.1

dependabot[bot] committed:
build(deps): bump @nestjs/common from 10.4.7 to 10.4.15

Bumps [@nestjs/common](https://github.com/nestjs/nest/tree/HEAD/packages/common) from 10.4.7 to 10.4.15.
- [Release notes](https://github.com/nestjs/nest/releases)
- [Commits](https://github.com/nestjs/nest/commits/v10.4.15/packages/common)

---
updated-dependencies:
- dependency-name: "@nestjs/common"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1597 from SciCatProject/dependabot/npm_and_yarn/nestjs/common-10.4.15

build(deps): bump @nestjs/common from 10.4.7 to 10.4.15

dependabot[bot] committed:
build(deps): bump mongoose from 8.8.4 to 8.9.2

Bumps [mongoose](https://github.com/Automattic/mongoose) from 8.8.4 to 8.9.2.
- [Release notes](https://github.com/Automattic/mongoose/releases)
- [Changelog](https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Automattic/mongoose/compare/8.8.4...8.9.2)

---
updated-dependencies:
- dependency-name: mongoose
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

github-actions[bot] committed:
Merge pull request #1595 from SciCatProject/dependabot/npm_and_yarn/mongoose-8.9.2

build(deps): bump mongoose from 8.8.4 to 8.9.2

Junjiequan commented:
@sbliven We are working on documentation for the dataset schema changes. I will publish this release now and add documentation link later when it's ready.
